From caefe6c9fa31b529b05124ee770af79e805c21cf Mon Sep 17 00:00:00 2001
From: Yasufumi Ogawa <ogawa.yasufumi@lab.ntt.co.jp>
Date: Tue, 16 Oct 2018 09:07:14 +0900
Subject: [PATCH 4/5] controller: move PORT_TYPES to spp_common

Move PORT_TYPES defining all of port types in SPP to spp_common module.

Signed-off-by: Yasufumi Ogawa <ogawa.yasufumi@lab.ntt.co.jp>
---
 src/controller/shell.py      |  8 ++++----
 src/controller/spp.py        |  5 +++--
 src/controller/spp_common.py | 12 +++++++-----
 3 files changed, 14 insertions(+), 11 deletions(-)

diff --git a/src/controller/shell.py b/src/controller/shell.py
index 383d2f1..9fce6f4 100644
--- a/src/controller/shell.py
+++ b/src/controller/shell.py
@@ -12,6 +12,7 @@ import os
 import re
 import readline
 from .shell_lib import common
+from . import spp_common
 from .spp_common import logger
 import subprocess
 
@@ -26,8 +27,6 @@ class Shell(cmd.Cmd, object):
     intro = 'Welcome to the spp.   Type help or ? to list commands.\n'
     prompt = 'spp > '
 
-    PORT_TYPES = ['phy', 'ring', 'vhost', 'pcap', 'nullpmd']
-
     PLUGIN_DIR = 'plugins'
     topo_size = '60%'
 
@@ -148,8 +147,9 @@ class Shell(cmd.Cmd, object):
             if re.match(ptn, id1) and re.match(ptn, id2):
                 pt1 = id1.split(delim)[0]
                 pt2 = id2.split(delim)[0]
-                if (pt1 in self.PORT_TYPES) and (pt2 in self.PORT_TYPES):
-                    return True
+                if (pt1 in spp_common.PORT_TYPES) \
+                        and (pt2 in spp_common.PORT_TYPES):
+                            return True
         return False
 
     def check_sec_cmds(self, cmds):
diff --git a/src/controller/spp.py b/src/controller/spp.py
index 99cdda3..5211ec9 100644
--- a/src/controller/spp.py
+++ b/src/controller/spp.py
@@ -20,8 +20,9 @@ def main(argv):
     args = parser.parse_args()
 
     try:
-        spp_ctl_cli = spp_ctl_client.SppCtlClient(args.bind_addr, args.api_port)
-        if spp_ctl_cli.is_server_running() == False:
+        spp_ctl_cli = spp_ctl_client.SppCtlClient(args.bind_addr,
+                                                  args.api_port)
+        if spp_ctl_cli.is_server_running() is False:
             print('Is not spp-ctl running, nor correct IP address?')
             exit()
         shell = Shell(spp_ctl_cli)
diff --git a/src/controller/spp_common.py b/src/controller/spp_common.py
index 809bee5..20c6bc8 100644
--- a/src/controller/spp_common.py
+++ b/src/controller/spp_common.py
@@ -5,6 +5,13 @@
 import logging
 import os
 
+SECONDARY_LIST = []
+
+PORT_TYPES = ['phy', 'ring', 'vhost', 'pcap', 'nullpmd']
+
+# Maximum num of sock queues for secondaries
+MAX_SECONDARY = 16
+
 # Setup logger object
 logger = logging.getLogger(__name__)
 # handler = logging.StreamHandler()
@@ -18,8 +25,3 @@ formatter = logging.Formatter(
 handler.setFormatter(formatter)
 logger.setLevel(logging.DEBUG)
 logger.addHandler(handler)
-
-SECONDARY_LIST = []
-
-# Maximum num of sock queues for secondaries
-MAX_SECONDARY = 16
-- 
2.13.1

